alert-common/src/main/java/com/synopsys/integration/alert/common/message/model/ComponentItem.java

```diff
@@ -96,13 +105,19 @@ public Long getNotificationId() {
         }
         for (final LinkableItem item : componentAttributes) {
             final String name = item.getName();
-            map.computeIfAbsent(name, ignored -> new ArrayList<>()).add(item);
+            map.computeIfAbsent(name, ignored -> new LinkedList<>()).add(item);
         }
         return map;
     }
 
+    @Override
+    public final int compareTo(final ComponentItem otherItem) {
```

Comment:
Rather than being Comparable, this could be a `createComparator` method that would only be used for processing.